lib/puppet/defaults.rb

```diff
@@ -32,6 +32,41 @@ def self.valid_file_checksum_types
       %w[sha256 sha256lite sha384 sha512 sha224 sha1 sha1lite md5 md5lite mtime ctime]
   end
 
+  def self.log_ca_migration_warning
+    urge_to_migrate = <<-UTM
+The cadir is currently in your puppet/ssl directory. This location will be
+deprecated and removed in a future version of puppet. Please run the the
+puppetserver ca tool to migrate it out to the puppetserver directory. Use
+`puppetserver ca migrate --help` for more info.
+UTM
+    Puppet.warn_once('deprecations',
+                     'CA migration message',
+                     urge_to_migrate,
+                     :default,
+                     :default)
+  end
+
+  def self.default_cadir
+    old_ca_dir = "#{Puppet[:ssldir]}/ca"
+    new_ca_dir = "/etc/puppetlabs/puppetserver/ca"
+
+    # If the file is a symlink, then we have run the migration tool
+    if File.exist?(old_ca_dir)
```

Comment:
Should that be "If the file is *not* a symlink..."?

Reply:
No, this is saying if it is, we have _already_ run it.

Reply:
ohhhh, I read that as "have to run", sorry..

Reply:
Oh, whoops, that comment was leftover from a my old code and shouldn't be there...

But to answer your question, the migration tool leaves behind a symlink at `/etc/puppetlabs/puppet/ssl/ca` pointing to `/etc/puppetlabs/puppetserver/ca`. So, if it is a symlink, just return what the symlink is pointing to. This catches a sort of edge case if someone had dropped in a manual a symlink manually, it will just read that link and use it as long as the target is not within the ssldir.